src/middlewared/middlewared/plugins/reporting/netdata/client.py

```diff
@@ -34,6 +35,9 @@ async def request(
     async def api_call(cls, resource: str, timeout: int = NETDATA_REQUEST_TIMEOUT, version: str = 'v1') -> dict:
         try:
             async with cls.request(resource, timeout, version) as resp:
-                return await resp.json()
+                output = ''
+                async for line in resp.content.iter_any():
+                    output += line.decode(errors='ignore')
+                return json.loads(output)
```

Comment:
Which exact operation did block the loop? The parsing of (seemingly) large JSON will still happen in the main loop

Reply:
@themylogin difference is when `json` method is called directly on the response - it reads the whole response content in a single go i.e
```
    async def json(
        self,
        *,
        encoding: Optional[str] = None,
        loads: JSONDecoder = DEFAULT_JSON_DECODER,
        content_type: Optional[str] = "application/json",
    ) -> Any:
        """Read and decodes JSON response."""
        if self._body is None:
            await self.read()
```

Whereas in this case we don't do that all at once and instead read it in chunks which results in us seeing the task being blocked anymore.